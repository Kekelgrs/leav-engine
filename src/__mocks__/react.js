const react = require('react');
module.exports = {...react, memo: x => x};

export default undefined;
