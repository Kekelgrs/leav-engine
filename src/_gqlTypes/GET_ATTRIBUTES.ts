/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import {AvailableLanguage, AttributeType, AttributeFormat, PermissionsRelation, ValueVersionMode} from './globalTypes';

// ====================================================
// GraphQL query operation: GET_ATTRIBUTES
// ====================================================

export interface GET_ATTRIBUTES_attributes_permissions_conf_permissionTreeAttributes {
    id: string;
    linked_tree: string | null;
    label: any | null;
}

export interface GET_ATTRIBUTES_attributes_permissions_conf {
    permissionTreeAttributes: GET_ATTRIBUTES_attributes_permissions_conf_permissionTreeAttributes[];
    relation: PermissionsRelation;
}

export interface GET_ATTRIBUTES_attributes_versions_conf {
    versionable: boolean;
    mode: ValueVersionMode | null;
    trees: string[] | null;
}

export interface GET_ATTRIBUTES_attributes {
    id: string;
    type: AttributeType;
    format: AttributeFormat | null;
    system: boolean;
    label: any | null;
    linked_library: string | null;
    linked_tree: string | null;
    multiple_values: boolean;
    permissions_conf: GET_ATTRIBUTES_attributes_permissions_conf | null;
    versions_conf: GET_ATTRIBUTES_attributes_versions_conf | null;
}

export interface GET_ATTRIBUTES {
    attributes: GET_ATTRIBUTES_attributes[] | null;
}

export interface GET_ATTRIBUTESVariables {
    lang?: AvailableLanguage[] | null;
    id?: string | null;
    label?: string | null;
    type?: (AttributeType | null)[] | null;
    format?: (AttributeFormat | null)[] | null;
    system?: boolean | null;
    multiple_values?: boolean | null;
    versionable?: boolean | null;
}
