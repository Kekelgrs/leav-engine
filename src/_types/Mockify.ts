// Used to mock any interface, turning all function properties to an optionnal mock
// Mockified object must be then passed to a function with a type assertion
export type Mockify<T> = {[P in keyof T]?: T[P] extends (...args: any) => any ? jest.Mock : T[P]};
