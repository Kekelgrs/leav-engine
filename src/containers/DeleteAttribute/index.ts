import DeleteAttribute from './DeleteAttribute';
export default DeleteAttribute;
