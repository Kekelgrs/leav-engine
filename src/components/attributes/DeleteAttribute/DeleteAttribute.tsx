import React from 'react';
import {useTranslation} from 'react-i18next';
import {DeleteAttributeMutation, deleteAttrQuery} from '../../../queries/attributes/deleteAttributeMutation';
import {getAttributesQueryName} from '../../../queries/attributes/getAttributesQuery';
import {GET_ATTRIBUTES_attributes_list} from '../../../_gqlTypes/GET_ATTRIBUTES';
import ConfirmedButton from '../../shared/ConfirmedButton';
import DeleteButton from '../../shared/DeleteButton';

interface IDeleteAttributeProps {
    attribute?: GET_ATTRIBUTES_attributes_list;
}

/* tslint:disable-next-line:variable-name */
const DeleteAttribute = (props: IDeleteAttributeProps): JSX.Element => {
    const {attribute} = props;
    const {t} = useTranslation();

    return !!attribute ? (
        <DeleteAttributeMutation mutation={deleteAttrQuery} refetchQueries={[getAttributesQueryName]}>
            {deleteAttr => {
                const onDelete = async () =>
                    deleteAttr({
                        variables: {attrId: attribute.id}
                    });

                const attrLabel =
                    attribute.label !== null ? attribute.label.fr || attribute.label.en || attribute.id : attribute.id;

                return (
                    <ConfirmedButton action={onDelete} confirmMessage={t('attributes.confirm_delete', {attrLabel})}>
                        <DeleteButton disabled={attribute.system} />
                    </ConfirmedButton>
                );
            }}
        </DeleteAttributeMutation>
    ) : (
        <></>
    );
};

export default DeleteAttribute;
