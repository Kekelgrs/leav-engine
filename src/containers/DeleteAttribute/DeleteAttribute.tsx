import {DataProxy} from 'apollo-cache';
import {TranslationFunction} from 'i18next';
import * as React from 'react';
import {translate} from 'react-i18next';
import {Confirm} from 'semantic-ui-react';
import DeleteButton from '../../components/DeleteButton';
import {DeleteAttributeMutation, deleteAttrQuery} from '../../queries/deleteAttributeMutation';
import {getAttributesQuery} from '../../queries/getAttributesQuery';
import {GET_ATTRIBUTES_attributes} from '../../_gqlTypes/GET_ATTRIBUTES';

interface IDeleteAttributeProps {
    attribute: GET_ATTRIBUTES_attributes;
    t: TranslationFunction;
}

interface IDeleteAttributeState {
    showConfirm: boolean;
}

class DeleteAttribute extends React.Component<IDeleteAttributeProps, IDeleteAttributeState> {
    constructor(props: IDeleteAttributeProps) {
        super(props);

        this.state = {
            showConfirm: false
        };
    }

    public render() {
        const {attribute, t} = this.props;
        const {showConfirm} = this.state;

        return (
            <DeleteAttributeMutation mutation={deleteAttrQuery} update={this._updateCache}>
                {deleteAttr => {
                    const onDelete = async (e: React.SyntheticEvent) => {
                        this._closeConfirm(e);
                        await deleteAttr({
                            variables: {attrId: attribute.id}
                        });
                    };

                    const attrLabel =
                        attribute.label !== null
                            ? attribute.label.fr || attribute.label.en || attribute.id
                            : attribute.id;

                    /**
                     * Prevent click on confirm modal from propagating to its parents
                     *
                     * @param e
                     */
                    const disableClick = (e: React.SyntheticEvent) => {
                        e.preventDefault();
                        e.stopPropagation();
                    };

                    return (
                        <div onClick={disableClick}>
                            <DeleteButton onDelete={this._openConfirm} />
                            <Confirm
                                open={showConfirm}
                                content={t('attributes.confirm_delete', {attrLabel})}
                                onCancel={this._closeConfirm}
                                onConfirm={onDelete}
                                cancelButton={t('admin.cancel')}
                                closeOnDocumentClick={false}
                                closeOnDimmerClick={false}
                            />
                        </div>
                    );
                }}
            </DeleteAttributeMutation>
        );
    }

    private _openConfirm = (e: React.SyntheticEvent) => {
        e.preventDefault();
        e.stopPropagation();
        this.setState({showConfirm: true});
    }
    private _closeConfirm = (e: React.SyntheticEvent) => {
        e.preventDefault();
        e.stopPropagation();
        this.setState({showConfirm: false});
    }

    private _updateCache(cache: DataProxy, {data: {deleteAttribute}}) {
        const cacheData: any = cache.readQuery({query: getAttributesQuery});
        cache.writeQuery({
            query: getAttributesQuery,
            data: {attributes: cacheData.attributes.filter(a => a.id !== deleteAttribute.id)}
        });
    }
}

export default translate()(DeleteAttribute);
