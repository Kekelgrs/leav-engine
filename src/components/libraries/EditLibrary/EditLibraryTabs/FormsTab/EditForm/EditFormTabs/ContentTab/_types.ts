import {DragObjectWithType} from 'react-dnd';
import {GET_ATTRIBUTES_attributes_list} from '../../../../../../../../_gqlTypes/GET_ATTRIBUTES';
import {FormElementTypes} from '../../../../../../../../_gqlTypes/globalTypes';
import {IKeyValue} from '../../../../../../../../_types/shared';

export const PLACEHOLDER_ID = '__placeholder__';

export enum UIElementTypes {
    DIVIDER = 'divider',
    FIELDS_CONTAINER = 'fields_container',
    TEXT_BLOCK = 'text_block',
    TABS = 'tabs'
}

export enum FieldTypes {
    TEXT_INPUT = 'input_field',
    DATE = 'date',
    CHECKBOX = 'checkbox',
    ENCRYPTED = 'encrypted',
    DROPDOWN = 'dropdown'
}

export enum DraggableElementTypes {
    RESERVE_LAYOUT_ELEMENT = 'RESERVE_LAYOUT_ELEMENT',
    ATTRIBUTE = 'ATTRIBUTE',
    FORM_ELEMENT = 'FORM_ELEMENT'
}

export interface IFormElement {
    id: string;
    order: number;
    type: FormElementTypes;
    uiElement: IUIElement;
    containerId: string;
    settings?: IKeyValue<any>;
    herited?: boolean;
}

export interface IUIElement {
    type: UIElementTypes | FieldTypes;
    component: JSX.Element;
    canDrop: (dropCandidate: IFormElement) => boolean;
    settings?: string[];
}

export interface IFormElementProps<T extends object> {
    elementData?: IFormElement;
    settings: T;
}

export interface ICommonFieldsSettings {
    label?: string;
    attribute?: GET_ATTRIBUTES_attributes_list;
}

export interface IFormElementPos {
    order: number;
    containerId: string;
}

export interface IFormBuilderDragObject<T extends IUIElement | IFormElement> extends DragObjectWithType {
    type: DraggableElementTypes;
    element: T;
    index: number;
    currentPos?: IFormElementPos;
    originPos?: IFormElementPos;
    dropAtPos?: IFormElementPos;
}
