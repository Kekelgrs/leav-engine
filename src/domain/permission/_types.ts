import {
    AppPermissionsActions,
    IPermissionsTreeTarget,
    LibraryPermissionsActions,
    PermissionsActions,
    PermissionTypes
} from '_types/permissions';
import {IQueryInfos} from '_types/queryInfos';
import {ITreeNode} from '_types/tree';

export interface IPermissionTarget {
    attributeId?: string;
    recordId?: string;
}

export interface IGetSimplePermissionsParams {
    type: PermissionTypes;
    applyTo: string | null;
    action: PermissionsActions;
    usersGroupId: string;
    permissionTreeTarget?: IPermissionsTreeTarget;
    ctx: IQueryInfos;
}

export interface IGetPermissionsByActionsParams {
    type: PermissionTypes;
    applyTo: string | null;
    actions: PermissionsActions[];
    usersGroupId: string;
    permissionTreeTarget?: IPermissionsTreeTarget;
    ctx: IQueryInfos;
}

export type PermByActionsRes = {[name: string]: boolean | null} | null;

export interface IGetPermissionByUserGroupsParams {
    type: PermissionTypes;
    action: PermissionsActions;
    userGroupsPaths: ITreeNode[][];
    applyTo?: string;
    permissionTreeTarget?: IPermissionsTreeTarget;
    ctx: IQueryInfos;
}

export interface IGetAppPermissionParams {
    action: AppPermissionsActions;
    userId: string;
    ctx: IQueryInfos;
}

export interface IGetHeritedAppPermissionParams {
    action: AppPermissionsActions;
    userGroupId: string;
    ctx: IQueryInfos;
}

export interface IGetLibraryPermissionParams {
    action: LibraryPermissionsActions;
    libraryId: string;
    userId: string;
    ctx: IQueryInfos;
}

export interface IGetHeritedLibraryPermissionParams {
    action: LibraryPermissionsActions;
    libraryId: string;
    userGroupId: string;
    ctx: IQueryInfos;
}

export interface IGetHeritedPermissionsParams {
    type: PermissionTypes;
    applyTo: string;
    action: PermissionsActions;
    userGroupId: string;
    permissionTreeTarget?: IPermissionsTreeTarget;
    ctx: IQueryInfos;
}

export interface IIsAllowedParams {
    type: PermissionTypes;
    action: PermissionsActions;
    userId: string;
    applyTo?: string;
    target?: IPermissionTarget;
    ctx: IQueryInfos;
}
