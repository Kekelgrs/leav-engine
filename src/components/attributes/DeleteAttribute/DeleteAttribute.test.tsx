import * as React from 'react';
import {MockedProvider} from 'react-apollo/test-utils';
import {create} from 'react-test-renderer';
import {GET_ATTRIBUTES_attributes} from 'src/_gqlTypes/GET_ATTRIBUTES';
import {Mockify} from 'src/_types/Mockify';
import DeleteAttribute from './DeleteAttribute';

describe('DeleteAttribute', () => {
    test('Snapshot test', async () => {
        const attr: Mockify<GET_ATTRIBUTES_attributes> = {
            label: null
        };
        const comp = create(
            <MockedProvider>
                <DeleteAttribute attribute={attr as GET_ATTRIBUTES_attributes} />
            </MockedProvider>
        );

        expect(comp).toMatchSnapshot();
    });
});
