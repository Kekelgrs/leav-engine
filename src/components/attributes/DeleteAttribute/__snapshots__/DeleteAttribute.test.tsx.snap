// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAttribute Snapshot test 1`] = `
<div
  onClick={[Function]}
>
  <button
    className="ui circular icon button delete"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    <i
      aria-hidden="true"
      className="trash icon"
    />
  </button>
</div>
`;
